### ISSUE TITLE
esm import throws error in node

### ISSUE THREAD

**Issue by detj** [humanwhocodes/env#14]
## Problem

On trying to esm import in node 14.16.0, getting the following error.

```
import { Env } from "@humanwhocodes/env";
         ^^^
SyntaxError: Named export 'Env' not found. The requested module '@humanwhocodes/env' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@humanwhocodes/env';
const { Env } = pkg;

    at ModuleJob._instantiate (internal/modules/esm/module_job.js:104:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:149:5)
    at async Loader.import (internal/modules/esm/loader.js:166:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
```

## Steps to reproduce

1. create an empty directory & cd into it - `mkdir env-try && cd env-try`
2. create a fresh package.json - `npm init --yes`
3. edit root package.json to add `"type": "module"`
3. install pkg - `npm install @humanwhocodes/env`
4. create new file with following contents - `touch index.js`

```js
import { Env } from "@humanwhocodes/env";

const env = new Env();
console.log(env);
```

6. On running using `node index` throws the above error.

## Notes

The package's package.json file correctly specifies ["export"](https://github.com/humanwhocodes/env/blob/0bf3a408dc7250516c19da0232a4476645320436/package.json#L11) field as per node's [conditional exports](https://nodejs.org/dist/latest-v14.x/docs/api/packages.html#packages_conditional_exports), but I'm not sure why this error gets thrown. Not sure if this issue is related to node's esm import support.

Although, adding `"type": "module"` in env's package.json solves this issue & the exported member is imported.
Am I missing something here, any idea?

```
node -v
v14.16.0

npm -v
7.6.0
```

**Comment by m-allanson** [comment]
I ran into this too. I believe this is because Node will treat any JavaScript file as CommonJS _unless_:

- the `package.json` has `type: module`, or
- the imported file extension is `.mjs`

Here's where I worked through it: https://github.com/m-allanson/node-esm-resolution. 

I think adding `type: module` would break the CommonJS imports? But changing the ESM file extension to `.mjs` should work (which is [option 1 in this article](https://2ality.com/2019/10/hybrid-npm-packages.html#option-1-(experimental%2C-needs-conditional-exports)%3A-esm-and-commonjs-are-both-bare-imports)).

**Comment by nzakas** [comment]
Thanks for the report. I'm not sure how I missed it, but I'll dig in and see if I can set up some automation to make sure this is caught with CI from now.

**Comment by nzakas** [comment]
This should be fixed in v2.1.3. Let me know if it's still not working.

**Comment by detj** [comment]
tested. works flawlessly.
